### ISSUE TITLE
Using ImportValue skips List type resource validation

### ISSUE THREAD

**Issue by anilkumarmyla** [aws-cloudformation/cfn-python-lint#1766]
*cfn-lint version: (`cfn-lint --version`)*
cfn-lint 0.40.0

*Description of issue.*
When we use [`Fn::ImportValue`](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-importvalue.html) for a resource key that needs type List, it accepts a scalar value and cfn-lint doesn't fail. But when we use scalar value directly it works as expected and fails. 

Following template needs `SubnetIds` and `SecurityGroupIds` as list of Strings (https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-lambda-function-vpcconfig.html), but using ImportValue masks the error and linting does not fail as expected

Template that should fail but succeeds
```yaml               
Resources:
  LambdaFunction:
    Type: 'AWS::Lambda::Function'
    Properties:
      Runtime: java8
      Code:
        S3Bucket: 'foo-bar'
        S3Key: 'foo-bar'
        S3ObjectVersion: 'foo-bar'
      Handler: 'com.foo.bar::handleRequest'
      Role: 'arn:aws:iam::123456789101:role/foo-bar'
      VpcConfig:
        SubnetIds: !ImportValue foobar1
        SecurityGroupIds: !ImportValue foobar2
```

Template that fails correctly
```yaml
Resources:
  LambdaFunction:
    Type: 'AWS::Lambda::Function'
    Properties:
      Runtime: java8
      Code:
        S3Bucket: 'foo-bar'
        S3Key: 'foo-bar'
        S3ObjectVersion: 'foo-bar'
      Handler: 'com.foo.bar::handleRequest'
      Role: 'arn:aws:iam::123456789101:role/foo-bar'
      VpcConfig:
        SubnetIds: 'foobar1'
        SecurityGroupIds: 'foobar2'

E3002 Property SubnetIds should be of type List for resource LambdaFunction
lambda-function.yaml:14:9

E3002 Property SecurityGroupIds should be of type List for resource LambdaFunction
lambda-function.yaml:15:9
```

**Comment by PatMyron** [comment]
From parsing the [Resource Specification](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/cfn-resource-specification.html) with a [script I wrote to find attribute types](https://github.com/aws-cloudformation/cfn-python-lint/pull/1659#discussion_r471847088), I think [`ImportValue`](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-importvalue.html) could resolve a `List`:
```bash
$ curl -s --compressed https://d1uauaxba7bl26.cloudfront.net/latest/gzip/CloudFormationResourceSpecification.json | jq '.ResourceTypes' | jq 'with_entries(.value |= .Attributes)' | grep -v ': null' | grep -Eo 'Type": ".*"' | sort | uniq -c | sort -nr
 582 Type": "String"
  27 Type": "List"
   7 Type": "Integer"
   4 Type": "Boolean"
   1 Type": "Map"
   1 Type": "Json"
...
```

**Comment by kddejong** [comment]
So if I have this right.  You can't export a List so no matter what ImportValue will not return a list.  `Template format error: The Value field of every Outputs member must evaluate to a String.` 

We should be able to say that this is a failure because we know ImportValue result is singular.

**Comment by anilkumarmyla** [comment]
thanks guys for the quick fix, much appreciated!
